endlessh.c

```diff
@@ -105,6 +106,12 @@ logsyslog(enum loglevel level, const char *format, ...)
     }
 }
 
+static long long random_delay(int delay, float jitter) {
+    long long tmp = delay*jitter;
+    tmp = delay+(rand()%(2*tmp))-tmp; //insecure rand() was used on purpose to avoid potential performance bottleneck
```

Comment:
This depends on undefined behavior if `jitter` is `0`.
Maybe add something like
```
if (jitter <= 0)
    return delay;
```